.changeset/blue-bags-warn.md

```diff
@@ -0,0 +1,7 @@
+---
+'@backstage/plugin-catalog-backend': minor
+'@backstage/plugin-catalog-node': minor
+'@backstage/plugin-catalog-backend-module-github': patch
+---
+
+Support adding location analyzers in new catalog analysis extension point and move `AnalyzeOptions` and `ScmLocationAnalyzer` types to `@backstage/plugin-catalog-node`
```

Comment:
nit: this could have been split into more than one changeset (for example, catalog-node didn't get the ability of adding location analyzers)

Reply:
hm but catalog-node does expose the new extension point related to this change though - I did split the -module-github package into it's own changeset however since that is mostly unrelated